tests/components/geo_json_events/test_geo_location.py

```diff
@@ -1,267 +1,170 @@
-            }
-            assert round(abs(float(state.state) - 25.5), 7) == 0
-
-            # Simulate an update - one existing, one new entry,
-            # one outdated entry
-            mock_feed_update.return_value = (
-                "OK",
-                [mock_entry_1, mock_entry_4, mock_entry_3],
-            )
-            async_fire_time_changed(hass, utcnow + DEFAULT_SCAN_INTERVAL)
-            await hass.async_block_till_done()
-
-            all_states = hass.states.async_all()
-            assert len(all_states) == 3
-
-            # Simulate an update - empty data, but successful update,
-            # so no changes to entities.
-            mock_feed_update.return_value = "OK_NO_DATA", None
-            async_fire_time_changed(hass, utcnow + 2 * DEFAULT_SCAN_INTERVAL)
-            await hass.async_block_till_done()
-
-            all_states = hass.states.async_all()
-            assert len(all_states) == 3
-
-            # Simulate an update - empty data, removes all entities
-            mock_feed_update.return_value = "ERROR", None
-            async_fire_time_changed(hass, utcnow + 3 * DEFAULT_SCAN_INTERVAL)
-            await hass.async_block_till_done()
-
-            all_states = hass.states.async_all()
-            assert len(all_states) == 0
-
-
-async def test_setup_with_custom_location(hass: HomeAssistant) -> None:
-    """Test the setup with a custom location."""
+async def test_setup_as_legacy_platform(hass: HomeAssistant) -> None:
+    """Test the setup with YAML legacy configuration."""
     # Set up some mock feed entries for this test.
-    mock_entry_1 = _generate_mock_feed_entry("1234", "Title 1", 2000.5, (-31.1, 150.1))
+    mock_entry_1 = _generate_mock_feed_entry("1234", "Title 1", 20.5, (-31.1, 150.1))
 
     with patch(
         "aio_geojson_generic_client.feed_manager.GenericFeed",
         wraps=GenericFeed,
     ) as mock_feed, patch(
-        "aio_geojson_client.feed.GeoJsonFeed.update"
-    ) as mock_feed_update:
-        mock_feed_update.return_value = "OK", [mock_entry_1]
-
-        with assert_setup_component(1, geo_location.DOMAIN):
-            assert await async_setup_component(
-                hass, geo_location.DOMAIN, CONFIG_WITH_CUSTOM_LOCATION
-            )
-            await hass.async_block_till_done()
-
-            # Artificially trigger update.
-            hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
-            # Collect events.
-            await hass.async_block_till_done()
+        "aio_geojson_client.feed.GeoJsonFeed.update",
+        return_value=("OK", [mock_entry_1]),
+    ):
+        assert await async_setup_component(hass, GEO_LOCATION_DOMAIN, CONFIG_LEGACY)
+        await hass.async_block_till_done()
 
-            all_states = hass.states.async_all()
-            assert len(all_states) == 1
+        # Artificially trigger update.
+        hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
+        await hass.async_block_till_done()
```

Comment:
You shouldn't need to fire this event anymore in the tests.

Reply:
OK, removed both instances.